### DISCUSSION TITLE
SITEURL and localhost - navigation links problem

### DISCUSSION THREAD

**Discussion by Jimmi08** [e107inc/e107#5217] [Q&A Core]
With SEF-URLs, you need full URLs in navigation. So I started to use something like this
`{SITEURL}/some-custom-link-in-sef-format `

It works on live site, but on localhost with e107 in subdirectory, it doesn't.
Link is always 
`http://localhost/some-custom-link-in-sef-format`

I always thought that value of this shortcode is from preferences. But it is set correctly:
![image](https://github.com/e107inc/e107/assets/5429548/fd8c4ef7-c80b-4e25-9398-1ee1b47faf1e)

So why this setting is ignored? Or how to correctly write URL in navigation - independent of an actual domain?  
Thanks

**Comment by tgtje** [comment]
@Jimmi08 i hope you can follow my thinking, and iff it makes sense,
/ (or domainname) on live server is an absolute path. (lets call it origin)
on local your in sub.. so its no longer origin but acts as a relative url path where localhost is origin.
No idea how to solve that, but maybe someone knows

**Comment by Vodhin** [comment]
`{SITEURL}` is just a placeholder for the domain name when parsed through a template. `localhost` _is_ a special domain name assigned to the IP address 127.0.0.1, and equates to the local computer, itself. It's built into the TCP/IP networking protocol for testing purposes. If you have the TCP/IP protocol installed and enabled, then "localhost" is a viable domain name. Test it by opening a command prompt (Start Menu > Windows System > Command Prompt) and typing "ping 127.0.0.1" or "ping localhost."

You may need to configure your firewall and/or install software to act as an actual webserver (to interpret commands the browser sends to a website in order to fetch the files to display)

**Reply by Jimmi08** [reply]
It is more complicated.  Localhost works normally. Subdirectory doesn't.

Siteurl is a preference, SITEURL constant is the value of this preference. {SITEURL} shortcode should have the same output as preference or constant.   If it is not set or it is "/", then you are right, but if I set that preference value, then it should use it. 

When it is not set, then rebase in .htacess should be the solution. But again, if I set it, then it should be respected.
